Added tests for pid-file deletion in DaemonMixIn

brejoc opened the pull request:
### What does this PR do?

This is a follow up on this PR:
https://github.com/saltstack/salt/pull/43366

Since we can get an OSError durin PIDfile deletion with non-root users, it
would make sense to also test for this. So here are the two test cases.
One with an OSError and the other one without.

### What issues does this PR fix or reference?

https://github.com/saltstack/salt/pull/43366

### Previous Behavior

No tests.

### New Behavior

Now there are tests.

### Tests written?

Yes

brejoc committed:
Added tests for pid-file deletion in DaemonMixIn

This is a follow up on this PR:
https://github.com/saltstack/salt/pull/43366

Since we can get an OSError durin PIDfile deletion with non-root users, it
would make sense to also test for this. So here are the two test cases.
One with an OSError and the other one without.

salt-jenkins commented:
@brejoc, thanks for your PR! By analyzing the history of the files in this pull request, we identified @vutny, @rallytime and @cro to be potential reviewers.

vutny commented on tests/unit/utils/parsers_test.py:
```diff
@@ -804,6 +805,62 @@ def setUp(self):
 
 
 @skipIf(NO_MOCK, NO_MOCK_REASON)
+class DaemonMixInTestCase(LogSettingsParserTests):
```
Does it really necessary to inherit from `LogSettingsParserTests`?
I see that you don't really need to do anything with logging... It would just execute all Runner tests once more.

brejoc replied:
I was unsure about this. Then I'll just get rid of it. Thanks!

vutny reviewed (changes requested):
Looks good in general, but I think we need to clean up few things.

brejoc commented on tests/unit/utils/parsers_test.py:
```diff
@@ -804,6 +805,62 @@ def setUp(self):
+        '''
+        with patch('os.unlink', MagicMock()) as os_unlink:
+            with patch('os.path.isfile', MagicMock(return_value=True)):
+	        with patch.object(self.mixin, 'info', MagicMock()):
```
Woops! That's was you get from using emacs instead of vim. ;)

brejoc commented on tests/unit/utils/parsers_test.py:
```diff
@@ -804,6 +805,62 @@ def setUp(self):
 @skipIf(NO_MOCK, NO_MOCK_REASON)
+class DaemonMixInTestCase(LogSettingsParserTests):
+    '''
+    Tests parsing Salt Master options
```
Sure, that slipped.

brejoc committed:
Fixed several issues with the test

* Removed redundant import.
* No longer inheriting from LogSettingsParserTests.
* Replaced test class description with somethin meaninful.
* Fixed identation.

I've also moved the class to the bottom, since all the classes inheriting
from LogSettingsParserTests are in the block above.

brejoc commented:
@vutny Thanks for the fast review. I think I've addressed your points in the last commit.

vutny commented:
There is just a small lint violation left: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/14782/violations/file/tests/unit/utils/parsers_test.py/

brejoc committed:
Fixed linting

Fix for "String format call with un-indexed curly braces".

brejoc commented:
@vutny Thanks, fixed it.

cachedout reviewed (approved):
Very nice! Thanks, @brejoc
